utils/db.ts

```diff
@@ -423,14 +424,25 @@ export async function getAnalyticsMetricsPerDay(
   metric: string,
   options?: Deno.KvListOptions,
 ) {
-  const iter = await kv.list<bigint>({ prefix: [metric] }, options);
-  const metricsValue = [];
-  const dates = [];
+  const limit = options?.limit ?? 0;
+  const metricsMap = new Map<string, number>(
+    Array.from({ length: limit })
+      .map<[string, number]>((_, index) => [
+        new Date(Date.now() - DAY * index).toISOString().split("T")[0],
+        0,
+      ])
+      .sort((a, b) => a[0].localeCompare(b[0])),
+  );
+  const iter = kv.list<bigint>({ prefix: [metric] }, options);
   for await (const res of iter) {
-    metricsValue.push(Number(res.value));
-    dates.push(String(res.key[1]));
+    if (metricsMap.has(String(res.key[1]))) {
+      metricsMap.set(String(res.key[1]), Number(res.value));
+    }
   }
-  return { metricsValue, dates };
+  const dates = Array.from(metricsMap.keys());
+  const metricsValue = Array.from(metricsMap.values());
+
+  return { dates, metricsValue };
 }
```

Comment:
thanks for the feedback.  
there was indeed quite a lot going on there.

i refactored the logic to be more aligned w/ existing functions which where expecting a `msAgo` argument.  
i also introduced some small helpers:  
- `getDatesSince` to get all dates since a given number of milliseconds ago
- `formatDate` to have a unified way of formatting dates

with the newly created tests it should be more predictable to see what's going on.  
however i'd still await the changes introduced in #296.